### ISSUE TITLE
[Secrets] Secrets are not detected in VS2019

### ISSUE THREAD

**Issue by duncanp-sonar** [SonarSource/sonarlint-visualstudio#3008]
### Description

The CI-built VSIX installs without error, but secrets are not detected in VS2019.
So far this issue has not been observed in VS2022.

Note: this failure does not occur on all VS2019 installations - on some machines the install works and secrets are detected.

### Notes
There is a MEF-composition error e.g. 
```
Error #2
Microsoft.VisualStudio.Composition.PartDiscoveryException: Failure while scanning type "SonarLint.VisualStudio.CloudSecrets.SecretsAnalyzer". ---> Microsoft.VisualStudio.Composition.PartDiscoveryException: Part discovery failed at member secretDetectors. ---> System.IO.FileNotFoundException: Could not load file or assembly 'SonarLint.Secrets.DotNet, Version=1.0.2.46905, Culture=neutral, PublicKeyToken=c5b62af9de6d7244' or one of its dependencies. The system cannot find the file specified.
   at System.Signature.GetSignature(Void* pCorSig, Int32 cCorSig, RuntimeFieldHandleInternal fieldHandle, IRuntimeMethodInfo methodHandle, RuntimeType declaringType)
   at System.Signature..ctor(IRuntimeFieldInfo fieldHandle, RuntimeType declaringType)
   at System.Reflection.RtFieldInfo.get_FieldType()
   at Microsoft.VisualStudio.Composition.ReflectionHelpers.GetMemberType(MemberInfo fieldOrPropertyOrType)
   at Microsoft.VisualStudio.Composition.AttributedPartDiscoveryV1.AddImportsFromMembers(PropertyInfo[] declaredProperties, FieldInfo[] declaredFields, TypeRef partTypeRef, IList`1 imports)
   --- End of inner exception stack trace ---
   at Microsoft.VisualStudio.Composition.AttributedPartDiscoveryV1.AddImportsFromMembers(PropertyInfo[] declaredProperties, FieldInfo[] declaredFields, TypeRef partTypeRef, IList`1 imports)
   at Microsoft.VisualStudio.Composition.AttributedPartDiscoveryV1.CreatePart(Type partType, Boolean typeExplicitlyRequested)
   at Microsoft.VisualStudio.Composition.PartDiscovery.CombinedPartDiscovery.CreatePart(Type partType, Boolean typeExplicitlyRequested)
   at Microsoft.VisualStudio.Composition.PartDiscovery.<>c__DisplayClass31_0.<CreateDiscoveryBlockChain>b__0(Type type)
   --- End of inner exception stack trace ---

```

The Fusion log shows that the secrets assembly from the NuGet package could not be found:
[SonarLint.Secrets.DotNet, Version=1.0.2.46905, Culture=neutral, PublicKeyToken=c5b62af9de6d7244.HTM.txt](https://github.com/SonarSource/sonarlint-visualstudio/files/8581981/SonarLint.Secrets.DotNet.Version.1.0.2.46905.Culture.neutral.PublicKeyToken.c5b62af9de6d7244.HTM.txt)

cf succesful Fusion log from VS2022: 
[VS2022_Succeeded_SonarLint.Secrets.DotNet, Version=1.0.2.46905, Culture=neutral, PublicKeyToken=c5b62af9de6d7244.HTM.txt](https://github.com/SonarSource/sonarlint-visualstudio/files/8582067/VS2022_Succeeded_SonarLint.Secrets.DotNet.Version.1.0.2.46905.Culture.neutral.PublicKeyToken.c5b62af9de6d7244.HTM.txt)

The secrets assembly is installed into a sub-directory of the extension because it is separately licensed (the sub-folder contains the DLL and a specific license file).

**Comment by duncanp-sonar** [comment]
Update: the bug didn't appear on most machines. There were no obvious distinguishing features of the machines it did reproduce on.

The current fix is to add the `secrets` subfolder to the VS binding path. Testing: there was one VM on which the bug repro-ed. I switched between two versions of the VSIX three times (manually installing, testing then uninstalling). The bug repro-ed in the bug without the binding path, and did not repro in the vsix with the binding path.  The fixed VSIX worked on three other dev machines.

**Possible underlying cause**
There are two assemblies, `SonarLint.VisualStudio.CloudSecrets.dll` (`CloudSecrets`) and `SonarLint.Secrets.DotNet.dll` (`Secrets`), laid out on disc as follows:

```
  [install root]\CloudSecrets.dll
  [install root]\secrets\Secrets.dll
```

Both assemblies are listed as MEF-components in the manifest.
VS can MEF-import both assemblies because the manifest specifies their locations.
`CloudSecrets` depends on `Secrets` i.e. when `CloudSecrets` loads, VS must be able to locate `Secrets` too.
Without the binding path attribute, Visual Studio will not search the `secrets` when looking for assembly dependencies.

Depending on the order in which the types are imported:
(A) `Secrets` is imported first, then `CloudSecrets` -> success. `CloudSecrets` loads ok because `Secrets` is already in memory.

(B) **[no binding path attribute]** `CloudSecrets` is imported first. VS tries to find `Secrets`. It is not in memory or on the search path -> failure with `Could not load file or assembly 'SonarLint.Secrets.DotNet...' or one of its dependencies.`

(C) **[with binding path attribute]** `CloudSecrets` is imported first. VS tries to find `Secrets`. It is on the binding path -> success.
